src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ViewPenaltiesController.java

```diff
@@ -84,15 +87,17 @@ public String getViewPenalties(@PathVariable String companyNumber,
             return ERROR_VIEW;
         }
 
-        model.addAttribute("outstanding", lateFilingPenalty.getOutstanding());
-        model.addAttribute("madeUpDate",
-                LocalDate.parse(lateFilingPenalty.getMadeUpDate(),
-                        DateTimeFormatter.ofPattern("uuuu-MM-dd", Locale.UK))
-                        .format(DateTimeFormatter.ofPattern("d MMMM uuuu", Locale.UK)));
+        DecimalFormat formatter = new DecimalFormat("#,###");
+        String formattedOutstanding = formatter.format(lateFilingPenalty.getOutstanding());
+
+        model.addAttribute("referenceTitle", penaltyNumber.startsWith("A") ? "Reference Number" : "Penalty Reference");
+        model.addAttribute("outstanding", formattedOutstanding);
         model.addAttribute("dueDate",
                 LocalDate.parse(lateFilingPenalty.getDueDate(),
                         DateTimeFormatter.ofPattern("uuuu-MM-dd", Locale.UK))
                         .format(DateTimeFormatter.ofPattern("d MMMM uuuu", Locale.UK)));
+        model.addAttribute("penaltyReference", penaltyNumber);
+        model.addAttribute("reasonForPenalty", PenaltyUtils.getPenaltyReason(penaltyNumber));
```

Comment:
Looks like this summary rows share some similarity with the new confirmation page being done in https://github.com/companieshouse/penalty-payment-web/pull/176

The approach for both probably needs to be shared via the new utility methods / properties being introduced for reason, amount.  FYI @klai-ch

Reply:
For SAN-125, it is using LateFilingPenalty while in SAN-24, it is using PayableLateFilingPenalty, so for some part it needs to take care if it can be share to the confirmation page or not.